### ISSUE TITLE
King-phisher client connection

### ISSUE THREAD

**Issue by Deensk** [securestate/king-phisher#69]
So I have both king-phisher and postgres services running, I was able to connect with the client with my own ssh credentials.  If the client is on a different machine, I can still create a new user and password through king-phisher server?
Any suggestion will be helpful. I think I’m missing something correct?  Thanks much.
Below is my server_config.yml
```
#
# King-Phisher Server Configuration
#

logging:
  file: /var/log/king-phisher.log
  level: warning
  console: false

server:
  # Bind address information
  address:
    host: 0.0.0.0
    port: 80

  # Authentication related information
  #authentication:
    # Require authenticating users to be a member of the specified group
    #group: king-phisher

  # A cookie to retrieve the client's IP from, useful when a proxy is in place
  #client_ip_cookie: X-Real-IP

  # The name of the cookie used to track users
  cookie_name: KPID

  # Specify an additional search path for server data files
  data_path: /opt/king-phisher-1.1.0

  # The SQLite3 database file to use, the folder needs write permissions
  # WARNING: SQLite3 is not fully supported as of version 0.1.6, see
  # https://github.com/securestate/king-phisher/wiki/Database#sqlite for more
  # details
  #database: sqlite:////var/king-phisher/king-phisher.db

  # The Postgres database to use
  database: postgresql://king_phisher:REDACTED@localhost/king_phisher

  # Fork the server process into the background
  fork: true

  geoip:
    # The location of the GeoLite2 City database
    database: /var/king-phisher/geoip.mmdb

  # Set a file to write the PID to
  pid_file: /var/run/king-phisher.pid

  # Require an ID associated with a campaign to load pages
  require_id: true

  rest_api:
    enabled: false
    # Networks which are authorized to use the REST API
    networks:
      - 10.0.0.0/8
      - 127.0.0.0/8
      - 172.16.0.0/12
      - 192.168.0.0/16
    # When null the token will be randomized at run time
    token: null

  # Configure the server header value
  #server_header: Apache/2.4.9 (Unix)

  # Set the opened timestamp on the first visit if it's null
  set_message_opened_on_visit: true

  # Drop privileges to the specified user
  setuid_username: nobody

  # SSL settings
  #ssl_cert: /path/to/ssl.crt
  #ssl_key: /path/to/ssl.key

  # Configure the image file name used to track opened emails
  tracking_image: email_logo_banner.gif

  # Use subfolders in the web_root which correspond to vhost names
  vhost_directories: false
  web_root: /var/www
```

**Comment by zeroSteiner** [comment]
The user authentication is done through PAM on Linux. What this means for you is that to create a new user account for King Phisher you just create a new Linux user account. It's important to ensure that the new user can log into the server using SSH (required for port forwarding) however they can have their shell set to /bin/nologin or equivalent.

**Comment by zeroSteiner** [comment]
I probably also should have mentioned that we have a [Server Authentication](https://github.com/securestate/king-phisher/wiki/Server-Authentication) wiki page which includes additional details you might be interested in. TLDR, you should be able to log in from another system where the client is installed so long as you can login from that system to the server with your user account and the King Phisher server is running.

**Comment by Deensk** [comment]
This is great thank you. Everything is running except when I try sending e-mail, it's saying it cannot connect. I am guessing I configured my SMTP server wrong correct? ssh running on port 22.
Connecting with localhost:22 instead of IP address make any difference in the connection dialog?

**Comment by zeroSteiner** [comment]
It's unclear which connection dialog you're referring to. Under preferences "SMTP Server" is the SMTP server that you would like to connect to. The connection can optionally be tunneled over SSH in which case the "SSH Server" field should be specified and SMTP Server will be forced to "localhost:25".

The SMTP Server should be an open relay or a server that you have access to (most commonly by authenticating to it) send email through to other domains.

**Comment by Deensk** [comment]
Sorry for the confusion. For king phisher configuration dialog box, I’m trying to figure out if all settings are correct.

**Comment by zeroSteiner** [comment]
Well what do you have it set to and what are you trying to connect to? Can you list the values of everything in the SMTP tab of preferences?

**Comment by Deensk** [comment]
SMTP tab preferences

SMTP Server =  Localhost:25
sending Rate =   45.0
connect with ssl  = off
Tunnel over SSh = on
SSH settings
SSH server  = d02ptrgrp:22
ssh username = jjones

**Comment by zeroSteiner** [comment]
Those look like the correct settings to use an SMTP server on d02ptrgrp over an SSH tunnel. Is that the desired configuration and is there an SMTP server running on that server?

If that is the desired configuration, then it would help to have the output of the DEBUG logs while you're trying to send a message. That would help to identify where the connection is failing.

**Comment by Deensk** [comment]
no, the SMTP is running on a different server and King phisher server can connect to it.

**Comment by zeroSteiner** [comment]
The server does not initiate the connection to the SMTP server, that's handled by the client. The client connects to the SMTP server, sends it's messages then informs the server about them.

Can the client connect to the SMTP server with SSH and initiate a port forward (because that's what it's configured to do)?

**Comment by Deensk** [comment]
yes, Client  connects but can't send a message. I will retry the settings.

**Comment by zeroSteiner** [comment]
Can you please provide the output from the logs while you are trying to send a message?

**Comment by Deensk** [comment]
Sorry, I have been a little busy. I think I'm getting close but can't figure out what is wrong with my smtp settings.
I set it as recommended but I think I’m missing web server URL part in the dialog box (config tab and required options)

This is the output log files that I'm getting.
2016-02-25 12:28:11,183 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:28:12,429 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/get
2016-02-25 12:28:12,434 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:28:13,692 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/get
2016-02-25 12:28:13,697 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:28:14,738 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:28:14,746 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:28:15,767 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:28:15,777 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:28:16,812 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:28:16,822 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:28:17,873 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:28:17,878 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:28:19,167 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:28:19,173 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:28:20,431 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/count
2016-02-25 12:28:20,436 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/count HTTP/1.1" 200 -
2016-02-25 12:30:31,098 AdvancedHTTPServer                                 INFO     running R                  PC method: /config/get
2016-02-25 12:30:31,098 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /config/get HTTP/1.1" 200 -
2016-02-25 12:30:32,346 AdvancedHTTPServer                                 INFO     running R                  PC method: /campaign/alerts/is_subscribed
2016-02-25 12:30:32,354 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /campaign/alerts/is_subscribed HTTP/1.1" 200 -
2016-02-25 12:30:33,594 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/get
2016-02-25 12:30:33,598 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:30:53,603 AdvancedHTTPServer                                 INFO     running R                  PC method: /config/set
2016-02-25 12:30:53,603 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /config/set HTTP/1.1" 200 -
2016-02-25 12:31:41,403 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/get
2016-02-25 12:31:41,409 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:32:00,170 AdvancedHTTPServer                                 INFO     running R                  PC method: /campaign/landing_page/new
2016-02-25 12:32:00,177 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /campaign/landing_page/new HTTP/1.1" 200 -
2016-02-25 12:32:57,065 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/get
2016-02-25 12:32:57,068 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/get HTTP/1.1" 200 -
2016-02-25 12:32:58,328 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:32:58,334 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:32:59,623 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/view
2016-02-25 12:32:59,633 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/view HTTP/1.1" 200 -
2016-02-25 12:33:00,902 AdvancedHTTPServer                                 INFO     running R                  PC method: /db/table/count
2016-02-25 12:33:00,907 AdvancedHTTPServer                                 INFO     localhost                  .localdomain "RPC /db/table/count HTTP/1.1" 200 -

**Comment by zeroSteiner** [comment]
From the logs it looks like you're starting to send an email. From the mail sender tab it should tell you that it's going through a few checks SPF, Web URL etc. Does it make it through those checks and tell you it sent the email? If it does not say it sent the email there should be some kind of error message and or stack trace.

**Comment by Deensk** [comment]
Yes, this is the output that I'm getting from the sender tab below. On "successfully send 0 emails" that should be 1 if any email is sent correct?
 
Checking the target URL... success, done.
Checking the SPF policy of target domain 'health.com'... done, no policy was found.
SPF policy result: N/A (No policy found)
File 'kingtest.txt' will be attached to sent messages.
  MD5:  dac8deb32e5cd0fcff6bdb76fd44212e
  SHA1: a046051515eb931d74b4af6cae4cea857db0cd25
Sending messages started at: Monday February 29, 2016 17:23:49
Connecting to SSH... done.
Connecting to SMTP server... done.
Finished sending emails, successfully sent 0 emails
Disconnected from the SMTP server
Disconnected from the SSH server

**Comment by zeroSteiner** [comment]
That's correct if there was one line in your CSV file and it was correctly formatted. Can you verify that your CSV file is in the correct order of `first name, last name, email address`. If the email address does not look correct or the first or last names are missing the line will be skipped, resulting in 0 emails being sent.

**Comment by Deensk** [comment]
It's sending but can't receive any mail from email address listed. In the output, the first email address is correct. Need to format HTML file?

Sending messages started at: Monday February 29, 2016 23:22:01
Connecting to SSH... done.
Connecting to SMTP server... done.
Sending email 1 of 4 wit   UID: DzhlfqIOwALarVi0 to samydeens@gmail.com
Sending email 2 of 4 with UID: VYX9EjusjnKmSWSF to deenskoolp@mail.com
Sending email 3 of 4 with UID: DmrreCLR4u9CoteE to ehell.tech@gmail.com
Sending email 4 of 4 with UID: wyDCFNRbj6xNeAh0 to more.bron@hotmail.com
Finished sending emails, successfully sent 4 emails
Disconnected from the SMTP server
Disconnected from the SSH server

**Comment by zeroSteiner** [comment]
If King Phisher says they're being sent they are making it to the configured SMTP server. Your messages are most likely being dropped as spam. If you're trying to send to GMAIL addresses, try authenticating to the GMAIL SMTP servers and sending from a legitimate address that you control. Regardless once the messages are sent their delivery are outside the control of King Phisher. You can also try to debug your SMTP server if you are using postfix or similar.

**Comment by Deensk** [comment]
You are right and everything seems to be sending now. I did receive some test emails that I sent early but the image won't show along with the text.
I tried something simple like this but not working
<img src="/images/CA2F.jpeg" alt="test image" />
   Is there any example or a document that I can look at to get it to work?

**Comment by Deensk** [comment]
You are right and everything seems to be sending now. I did receive some test emails that I sent early but the image won't show along with the text.
I tried something simple like this but not working
(<img src="/images/CA2F.jpeg" alt="test image" />)
   Is there any example or a document that I can look at to get it to work?

**Comment by zeroSteiner** [comment]
You can embed images two different ways, one involves them being stored on the server and the other is to include them "inline". Right click in the editor tab and select insert inline image to do it that way, otherwise ensure that the images URL on the server is correct and you're passing the uid if the `require_id` setting is enabled (which is the default). An example would be something like `<img src="http://mykingphisher.lan/lolcat.png?id={{ uid }}">`.

If you're still having trouble please tell me which method you're trying to use to embed the image, normal or inline.

**Comment by zeroSteiner** [comment]
I realized that there was no wiki page explaining the two different ways on how to embed images into messages so I created a new "Creating Message Content" page with a section on [Embedding Images Into Messages](https://github.com/securestate/king-phisher/wiki/Creating-Message-Content#embedding-images-into-messages).

If you have additional questions that should hopefully help you. If you don't need additional help feel free to close this issue out.
